src/main/java/org/openstreetmap/atlas/checks/validation/points/InvalidMiniRoundaboutCheck.java

```diff
@@ -0,0 +1,149 @@
+ * Edges as potentially tagged incorrectly.
+ *
+ * @author nachtm
+ */
+public class InvalidMiniRoundaboutCheck extends BaseCheck<Long>
+{
+
+    private static final long DEFAULT_VALENCE = 6;
+    private static final String MINIMUM_VALENCE_KEY = "valence.minimum";
+    private static final String OTHER_EDGES_INSTRUCTION = "This Mini-Roundabout Node ({0,number,#})"
+            + " has {1, number,#} connecting car-navigable Edges. Consider changing this.";
+    private static final String TWO_EDGES_INSTRUCTION = "This Mini-Roundabout Node ({0,number,#}) "
+            + "has 2 connecting car-navigable Edges. Consider changing this to highway=TURNING_LOOP or "
+            + "highway=TURNING_CIRCLE.";
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList(TWO_EDGES_INSTRUCTION,
+            OTHER_EDGES_INSTRUCTION);
+    private static final DirectionTag[] VALID_DIRECTIONS = { DirectionTag.CLOCKWISE,
+            DirectionTag.ANTICLOCKWISE };
+    private final long valenceMinimum;
+
+    /**
+     * Construct an InvalidMiniRoundaboutCheck with the given configuration values.
+     *
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public InvalidMiniRoundaboutCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.valenceMinimum = this.configurationValue(configuration, MINIMUM_VALENCE_KEY,
+                DEFAULT_VALENCE);
+    }
+
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return object instanceof Node
+                && Validators.isOfType(object, HighwayTag.class, HighwayTag.MINI_ROUNDABOUT);
+    }
+
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Node node = (Node) object;
+        final Collection<Edge> carNavigableEdges = this.getCarNavigableEdges(node);
+        final long valence = carNavigableEdges.size();
+        final Optional<CheckFlag> result;
+
+        // If this Node is a turnaround, we always want to flag it.
+        if (this.isTurnaround(carNavigableEdges))
+        {
+            result = Optional.of(this.flagNode(node, carNavigableEdges,
+                    this.getLocalizedInstruction(0, node.getOsmIdentifier())));
+        }
+        // Otherwise, if there is not a direction tag, and the valence is less than valenceMinimum,
+        // we want to flag it.
+        else if (!Validators.isOfType(node, DirectionTag.class, VALID_DIRECTIONS)
+                && valence < valenceMinimum && valence > 0)
+        {
+            result = Optional.of(this.flagNode(node, carNavigableEdges,
+                    this.getLocalizedInstruction(1, node.getOsmIdentifier(), valence)));
+        }
+        // Otherwise, this mini-roundabout is probably legitimate. Don't flag it.
+        else
+        {
+            result = Optional.empty();
```

Comment:
Nit: You could potentially initialize this to empty to begin with and only update in case of flagging action.